vyper/venom/venom_to_assembly.py

```diff
@@ -118,6 +119,13 @@ def apply_line_numbers(inst: IRInstruction, asm) -> list[str]:
     return ret  # type: ignore
 
 
+def _is_cleanup_needed(fn: IRFunction) -> bool:
+    for bb in fn.get_basic_blocks():
+        if bb.is_terminated and bb.instructions[-1].opcode == "ret":
```

Comment:
`is_terminated` and then checking for `ret` seems superfluous. Just a check if `instructions` not empty.